src/datasets/open_weiboscope.py

```diff
@@ -0,0 +1,185 @@
+    data_sets['train'], data_sets['test'] = ow_files[:split_on], ow_files[split_on:]
+    logger.debug(data_sets)
+    for table_path in data_sets[which_set]:
+        with codecs.open(table_path, "r", encoding="utf-8") as f:
+            logging.debug("In file {}".format(table_path))
+            for line in unicode_csv_reader(f):
+                try:
+                    records_split = line
+                    if len(records_split) != 11:
+                        raise  BadRecordException("Comma split error on mid={} in"
+                                         "file {} (len of record: {})".format(
+                                            records_split[0], 
+                                            os.path.basename(table_path),
+                                            len(records_split)))
+            
+                    # text on field 6, deleted on field 9. screen out retweets (field 1) 
+                    if records_split[1] == '':
+                        if form=='hanzi':
+                            record_txt, sentiment = enforce_length(
+                                records_split[6], min_length, max_length, 
+                                pad_out), records_split[9] != '' 
+                            yield record_txt, sentiment
+                        elif form=='pinyin':
+                            record_txt, sentiment = enforce_length(
+                                romanize_tweet(records_split[6]), min_length, 
+                                max_length, pad_out), records_split[9] != '' 
+                            yield record_txt, sentiment
+                        elif form=='onehot':
+                            record_txt, sentiment = enforce_length(
+                                romanize_tweet(records_split[6]), min_length, 
+                                max_length, pad_out), records_split[9] != '' 
+                            yield text_to_one_hot(record_txt, vocabulary), sentiment
+                        else:
+                            raise Exception("Unknown form '{}' (should be 'hanzi', "
+                                            "'pinyin', or 'onehot')".format(form))
+
+                except TextTooShortException:
+                    continue
+                except BadRecordException:
+                    continue
+                except IndexError:
+                    continue
+                except UnicodeEncodeError:
+                    continue
+
+                except GeneratorExit:
+                    return
+
+def text_to_one_hot(txt, vocabulary=vocabulary):
+    # setup the vocabulary for one-hot encoding
+    vocab_chars = set(list(vocabulary))
+
+    # create the output list
+    chars = list(txt)
+    categorical_chars = pd.Categorical(chars, categories=vocab_chars)
+    vectorized_chars = np.array(pd.get_dummies(categorical_chars))
+    return vectorized_chars
+
+def romanize_tweet(txt):
```

Comment:
reading through the code, it looks like this method first segments the text to hanzi tokens, then converts those tokens to pinyin, and returns a space-separated string of pinyin "romanized" tokens.  that took a bit of reading to figure out, so it might help newcomers to have a comment block explaining this method